integration-tests/tests/src/tests/mixed.ts

```diff
@@ -0,0 +1,313 @@
+      const date = new Date();
+
+      const data: ISingleSchema = this.realm.write(() =>
+        this.realm.create<ISingleSchema>(SingleSchema.name, { a: oid }),
+      );
+      expect(typeof data.a === typeof oid, "should be the same type ObjectId");
+      expect(String(data.a)).equals(oid.toString(), "should have the same content");
+
+      this.realm.write(() => (data.a = uuid));
+      expect(typeof data.a === typeof uuid, "should be the same type UUID");
+      expect(String(data.a)).equals(uuid.toString(), "should have the same content UUID");
+
+      this.realm.write(() => (data.a = d128));
+      expect(String(data.a)).equals(d128.toString(), "Should be the same Decimal128");
+
+      this.realm.write(() => (data.a = 12345678));
+      expect(data.a).equals(12345678, "Should be the same 12345678");
+
+      this.realm.write(() => ((data.a = null), "Should be the same null"));
+      expect(data.a).equals(null);
+
+      this.realm.write(() => ((data.a = undefined), "Should be the same null"));
+      expect(data.a).equals(null);
+    });
+    it("wrong type throws", function (this: RealmContext) {
+      expect(() => {
+        this.realm.write(() => this.realm.create(SingleSchema.name, { a: Object.create({}) }));
+      }).throws(Error, "Only Realm instances are supported.");
+    });
+  });
+
+  describe("Nested types", () => {
+    openRealmBefore({ schema: [SingleSchema, VertexSchema, MixNestedSchema] });
+    it("support nested types", function (this: RealmContext) {
+      this.realm.write(() => {
+        const r = this.realm.create(VertexSchema.name, { a: 1, b: 0, c: 0 });
+        const r2 = this.realm.create(VertexSchema.name, { a: 0, b: 1, c: 0 });
+        const r3 = this.realm.create(VertexSchema.name, { a: 0, b: 0, c: 1 });
+
+        this.realm.create(SingleSchema.name, { a: r, b: r2, c: r3, d: null });
+      });
+
+      let data: ISingleSchema = this.realm.objects<ISingleSchema>(SingleSchema.name)[0];
+
+      expect((data.a as IVertexSchema).a).equals(1, "Should be equal to 1");
+      expect((data.a as IVertexSchema).b).equals(0, "Should be equal 0");
+      expect((data.a as IVertexSchema).c).equals(0, "Should be equal 0");
+
+      expect((data.b as IVertexSchema).a).equals(0, "Should be equal 0");
+      expect((data.b as IVertexSchema).b).equals(1, "Should be equal 1");
+      expect((data.b as IVertexSchema).c).equals(0, "Should be equal 0");
+
+      expect((data.c as IVertexSchema).a).equals(0, "Should be equal 0");
+      expect((data.c as IVertexSchema).b).equals(0, "Should be equal 0");
+      expect((data.c as IVertexSchema).c).equals(1, "Should be equal 1");
+
+      this.realm.write(() => {
+        const r = this.realm.create(MixNestedSchema.name, { a: 0, b: -1 });
+        const r1 = this.realm.create(MixNestedSchema.name, { a: 1, b: 0 });
+        this.realm.create(SingleSchema.name, { a: r, b: r1 });
+      });
+
+      data = this.realm.objects<ISingleSchema>(SingleSchema.name)[1];
```

Comment:
I'd also suggest naming this differently and use two `const` instead of a `let`.